### PR TITLE
Add a new flag --no-processes to hide process information

### PR DESCRIPTION
**Pull request by doncamilom**
Hey I find this useful for my own application, also seems to solve #101 I think :)

### CODE REVIEW

**Comment by wookayin** [comment]
Hello, thanks for your contribution!

```
`-n`, `--no-processes`    : Hide memory data for individual processes.
```

Actually the name of the flag doesn't fully explain or match its description.  It more sounds like it would completely disable the process information.

FYI, this can be also seen as a special case of #51 (more general solution) -- users may want to have more fine-grained control of process output; we may want to avoid having tons of command-line flags.

**Comment by wookayin** [comment]
I see. So it tries to hide all the process information. Suggestions:

- I think it'd better to not add the shorthand `-n` flag, i.e. let's use `--no-processes` only.
- `Hide memory data for individual processes.`  this description doesn't match your intention. Can you please fix the documentation?
- Please update README.md, the usage section accordingly.
- I would prefer not adding the `|` separator at the end of line, when process information is hidden.
- Please add tests, in `test_args_commandline`. (see https://github.com/wookayin/gpustat/blob/master/gpustat/test_gpustat.py#L308)

**Comment by doncamilom** [comment]
Thank you for your review! 
Hope the documentation is better now.

File: gpustat/test_gpustat.py [hunk 1]
```diff
@@ -200,6 +200,11 @@ def _MockedProcess(pid):
 [2] GeForce RTX 2       | 71°C, 100 %,  ?? % (E:  ?? %  D:  ?? %),  250 /  ?? W |     0 / 12189 MB | (Not Supported)
 """.splitlines())  # noqa: E501
 
+MOCK_EXPECTED_OUTPUT_NO_PROCESSES = os.linesep.join("""\
+[0] GeForce GTX TITAN 0 | 80°C,  76 % |  8000 / 12287 MB
+[1] GeForce GTX TITAN 1 | 36°C,   0 % |  9000 / 12189 MB
+[2] GeForce RTX 2       | 71°C,  ?? % |     0 / 12189 MB | (Not Supported)
```
Comment (wookayin) [inline gpustat/test_gpustat.py:206]: When `--no-processes` is given, `| (Not Supported)` should not appear as well.

File: gpustat/core.py [hunk 2]
```diff
@@ -326,15 +327,21 @@ def full_process_info(p):
...
-        else:
+        elif not no_processes:
             for p in processes:
                 reps += ' ' + process_repr(p)
                 if show_full_cmd:
                     full_processes.append(os.linesep + full_process_info(p))
+
+        # Remove trailing | if no process information is shown.
+        if no_processes and reps[-1]=="|":  
+            reps = reps[:-2]
```
Comment (wookayin) [inline gpustat/core.py:338]: Sorry to nitpick this but please follow the conventional styles. Trailing spaces should be stripped also. In addition, if you want to strip like `reps[:-2]` then `.endswith` check should be with `" |"`, not `"|"`.

Comment (doncamilom) [reply gpustat/core.py]: Great, removed the stripping altogether, and now instead added the " |" conditionally.

File: gpustat/core.py [hunk 3]
```diff
@@ -181,6 +181,7 @@ def print_to(self, fp,
                  show_codec="",
                  show_power=None,
                  gpuname_width=None,
+                 no_processes=False,
```
Comment (wookayin) [inline gpustat/core.py:184]: We've got some conflicts. Can you please move this argument near to `show_full_cmd` (for better locality), and *squash* all the commits without merge-with-up-stream merge-commits? Thank you.

**Review by wookayin — CHANGES_REQUESTED** [review]
We've got some conflicts. Can you please move some arguments and squash all the commits without merge-with-up-stream merge-commits? Thank you.

File: gpustat/core.py (hunk 3 above)
Comment (doncamilom) [reply gpustat/core.py]: squash merging is activated, lmk if it works :)

Comment (wookayin) [reply gpustat/core.py]: I can do squash merging once LGTM, but it'd be also helpful for you to learn an even better github PR workflow with rebasing and squashing commits. Anyway thanks for the update!

**Review by wookayin — APPROVED** [review]

**Comment by wookayin** [comment]
Thank you, I've merged this.

**Pull request merged**